src/lib/components/tabs/QTabs.svelte

```diff
@@ -1,67 +1,103 @@
     round: QTabsProps["round"] = false,
-    userClasses: QTabsProps["userClasses"] = undefined,
-    userStyles: QTabsProps["userStyles"] = undefined;
-  export { userClasses as class, userStyles as style };
+    userClasses: QTabsProps["userClasses"] = "";
+  export { userClasses as class };
 
-  // Hide indicator until it's mounted
-  let hidden = true;
+  let qTabs: HTMLElement;
 
-  onMount(() => {
-    setTimeout(() => {
-      hidden = false;
-    }, 200);
+  const qTabStore = writable<QTabStore>({
+    value,
+    variant,
+    previousEl: null,
+    activeEl: null,
+    utils: { size: 0, position: 0 },
   });
 
-  /** Counting QTab children number */
-  let QTabCount = 0;
-  setContext("QTabCount", {
-    index: () => {
-      QTabCount += 1;
-      return QTabCount;
-    },
-  });
+  /* const indicatorWidth = derived(qTabStore, ($store) => {
+    return variant === "primary" ? `calc(${$store.size}px + 8px)` : `${$store.size}px`;
+  }); */
 
-  let activeTabStore = writable({ name: value, index: 0, size: 0, position: 0 });
-  $: activeTabStore.update(($active) => {
-    $active.name = value;
+  // Update the store when "value" changes programmatically
+  $: qTabStore.update(($store) => {
+    $store.value = value;
 
-    return $active;
+    return $store;
   });
-  setContext("activeTab", activeTabStore);
-  setContext("variant", variant);
+  // Update "value" when the store changes from QTab
+  $: value = $qTabStore.value;
 
-  $: value = $activeTabStore.name;
+  setContext("qTabStore", qTabStore);
 
-  $: classes = createClasses([variant, round && "rounded", hidden && "hidden-indicator"], {
-    component: "q-tabs",
-    userClasses,
-  });
+  $: if ($qTabStore.activeEl !== null) {
+    const {
+      previousEl,
+      activeEl,
+      utils: { position, size },
+      variant: storeVariant,
+    } = $qTabStore;
+    const tabsSize = storeVariant === "vertical" ? qTabs.offsetHeight : qTabs.offsetWidth;
+    const tabSize = size / tabsSize;
 
-  let indicatorWidth = derived(activeTabStore, ($activeTabStore) => {
-    return variant === "primary"
-      ? `calc(${$activeTabStore.size}px + 8px)`
-      : `${$activeTabStore.size}px`;
-  });
+    const child = variant === "primary" ? activeEl.firstElementChild as HTMLDivElement : { offsetLeft: 0, offsetWidth: 0 }
+    const previousChild = variant === "primary" ? previousEl?.firstElementChild as HTMLDivElement : { offsetLeft: 0, offsetWidth: 0 }
+
+    if (previousEl === null) {
+      // Position initial indicator
+      qTabs.style.setProperty("--indicator-size", `${tabSize}`);
+      qTabs.style.setProperty("--indicator-position", `${position}px`);
+    } else {
+      // Position indicator on tab change
+      const comparePositions = movementDirection(previousEl, activeEl);
+
+      let transitionSize;
+      if (comparePositions === "next") {
+        // New tab is after the previous one
+        transitionSize =
+          storeVariant === "vertical"
```

Comment:
This seems to be the same algorithm as in the `else` branch, only with different variable order, if I'm not mistaken. If so, this should be refactored into a function that takes two parameters.